packages/theme-context/src/ThemeContext/createThemeProxy.js

```diff
@@ -1,23 +1,61 @@
 export default function createThemeProxy(theme) {
-  return new Proxy(theme, {
-    get: (obj, propertyName) => {
-      const value = obj[propertyName];
-      const isValidProperty = Object.prototype.hasOwnProperty.call(
-        obj,
-        propertyName
-      );
-
-      if (!isValidProperty) {
-        /* eslint-disable-next-line no-console */
-        console.error(`Role ${propertyName.toString()} does not exist`);
-        return null;
+  const isPresent = (obj, prop) =>
+    Object.prototype.hasOwnProperty.call(obj, prop);
+
+  const resolvedRolesProxy = roles => {
```

Comment:
Would it make sense to move this outside the scope of the `createThemeProxy` function so we're not creating a new function (and a new scope) every time it's called? Would also allow you to test it explicitly.